### PR TITLE
implicitly restarted arnoldi method for JaxBackend

### PR DESCRIPTION
**Pull request by mganahl**
depends on #641

### CODE REVIEW

**Comment by Thenerdstation** [comment]
resolve the conflicts.

**Comment by Thenerdstation** [comment]
Let's fix the comments and pull this in?

**Comment by mganahl** [comment]
uh, sorry I completely missed your comments!! I'll fix it

File: tensornetwork/backends/jax/jitted_functions.py [hunk 1]
```diff
@@ -303,3 +305,213 @@ def check_thresh(check_vals):
...
+      fm = fm.astype(dtype)
+
+    while (it < maxiter) and (not converged):
+      evals, _ = jax.numpy.linalg.eig(Hm)
+      krylov_vectors, H, fk = shifted_QR(Vm, Hm, fm, evals, numeig, p, _which)
+      v0 = jax.numpy.reshape(fk, initial_state.shape)
+      # restart
+      Vm_tmp, Hm_tmp, _, converged = arnoldi_fact(matvec, args, v0,
+                                                  krylov_vectors, H, numeig,
+                                                  num_krylov_vecs, eps)
```
[the remark this answers is not in the document]
Comment (mganahl) [reply tensornetwork/backends/jax/jitted_functions.py:503]: usually yes, but this throws an error if the function is jitted

**Pull request merged**